client/src/app/site/base/base-controller.ts

```diff
@@ -1,23 +1,20 @@
 import { BaseModel, ModelConstructor } from '../../domain/models/base/base-model';
 import { BaseViewModel } from './base-view-model';
-import { ControllerServiceCollectorService } from '../services/controller-service-collector.service';
 import { Observable } from 'rxjs';
 import { BaseRepository } from '../../gateways/repositories/base-repository';
 import { Id } from '../../domain/definitions/key-types';
 import { TranslateService } from '@ngx-translate/core';
 import { Directive } from '@angular/core';
+import { AppInjector } from 'src/app/openslides-main-module/services/app-injector.service';
 
 @Directive()
 export abstract class BaseController<V extends BaseViewModel, M extends BaseModel> {
-    public get translate(): TranslateService {
-        return this.controllerServiceCollector.translate;
-    }
+    public translate: TranslateService;
 
-    public constructor(
-        protected controllerServiceCollector: ControllerServiceCollectorService,
-        protected baseModelCtor: ModelConstructor<M>,
-        protected repo: BaseRepository<V, M>
-    ) {}
+    public constructor(protected baseModelCtor: ModelConstructor<M>, protected repo: BaseRepository<V, M>) {
+        const injector = AppInjector.getInjector();
+        this.translate = injector.get(TranslateService);
```

Comment:
Here we have to check, if this is sufficient to get the translations extracted from the TypeScript files. To check this, run `npm run extract` and check, if there are some changes. This might be related to #1112.

Reply:
You are right, this destroys the extractability of strings. We have multiple options now as I see it:
- use another extraction tool (probably not practical)
- fork https://github.com/biesbjerg/ngx-translate-extract and implement a fix for this (maybe some more work, but probably the cleanest solution - unfortunately, the repo seems to be no longer maintained, which means on the other hand that we do not need to update the fork anytime soon)
- explicitly override the `translate` variable in all subclasses/use the traditional DI for the injection of the `TranslateService` (also some work, but very redundant one, but the solution is not as clean as the fork one)